### PR TITLE
feat: implement subscription filter

### PR DESCRIPTION
**Pull request by Aenimus**
## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

**Review by StarpTech — CHANGES_REQUESTED** [review]
Only one point other than that great work!

File: router/config.yaml [hunk 1]
```diff
@@ -3,4 +3,18 @@
 # See pkg/config/config.go for the full list of configuration options.
 # This file is used to debugging purposes only.
 
-version: "1"
\ No newline at end of file
+version: "1"
```
[the remark this answers is not in the document]
Comment (Aenimus) [reply router/config.yaml:6]: It's just removed the newline.

**Review by jensneuse — APPROVED** [review]
I think you've implemented unnecessarily overlapping tests.
The filter is transport agnostic as it's implemented in the engine.
If we change something in the future, we'd have to change all tests for all transports, but we're not getting any additional value from these overlapping tests.
If I'm not missing something, I'd prefer to have as few tests as possible and as many as they are adding additional test coverage, which is not the case here.

That said, looks very good!

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**